### ISSUE TITLE
Dynamic Program Service Name

### ISSUE THREAD

**Issue by MinePro120** [ChristopheJacquet/PiFmRds#103]
Hi,
I was wondering whether it is possible to display multiple -ps texts. Like the ones real radio stations transmit eg :
Rock 
Radio
97,6
Stay
Tuned

**Comment by Miegl** [comment]
You can use a FIFO pipe for that. 
```
mkfifo rds_ctl
sudo ./pi_fm_rds -ctl rds_ctl
```
```
cat >rds_ctl
PS Rock
PS Radio
```

**Comment by MinePro120** [comment]
Thank you very much. As I am a complete noob at this, do I just have to run those commands from the PiFmRds directory?

**Comment by Miegl** [comment]
Just run these commands in the directory with pi_fm_adv.

**Comment by MinePro120** [comment]
I got it. Changing the ps in real time is not what I am looking for. I want it to be automatic and loop the same ps's again and again without me having to do anything. Is it possible?

**Comment by Miegl** [comment]
https://bash.cyberciti.biz/guide/Main_Page
https://www.cyberciti.biz/faq/bash-for-loop/

**Comment by Miegl** [comment]
https://bash.cyberciti.biz/guide/Main_Page
https://www.cyberciti.biz/faq/bash-for-loop/

**Comment by collinator24** [comment]
I wanted to do this as well. Did anyone figure it out?

**Comment by collinator24** [comment]
did you figure it out?

**Comment by ChristopheJacquet** [comment]
Currently you have to send a "PS" command to the FIFO each time you want to change the PS. You can do a loop, e.g. in bash, to produce a cyclic effect.
I added the "enhancement" label: if I have time I may add this feature directly to Pi-FM-RDS.

**Comment by collinator24** [comment]
That's cool. For some reason, when I made my script it would stop working
after the first one. It doesn't let me cat the file when the pifmrds is
running. If I cat the file, then run it it works but because my script does
it, sleeps, then does a different command it doesn't work. Maybe I just
have bad luck.

**Comment by MinePro120** [comment]
I created the script some time ago, it worked fine.

**Comment by WisseHesNL** [comment]
Can you send the script?

**Comment by MinePro120** [comment]
#!/bin/bash

while true
do
	echo 'PS Tempo FM' | cat >rds_ctl
	sleep 5
	echo 'PS Quality' | cat >rds_ctl
	sleep 5
	echo 'PS Music' | cat >rds_ctl
	sleep 5
	echo 'PS 108 MHz' | cat >rds_ct
	sleep 5
	echo 'PS Stay' | cat >rds_ctl
	sleep 5
	echo 'PS Tuned!' | cat >rds_ctl
	sleep 5
	echo 'PS '$(date +"%H:%M") | cat >rds_ctl
	sleep 5
        echo 'PS '$(date +"%d/%m/%y") | cat >rds_ctl
	sleep 5
done
exit 0

There you go. I've added some extras too.

**Comment by WisseHesNL** [comment]
Thanks!

**Comment by WisseHesNL** [comment]
Hmm, apparently `echo 'PS radio' | cat >rds_ctl` only works when you restart PiFmRds everytime you change it, but 
```
pi@pi-fm:~/PiFmRds/src $ cat >rds_ctl
PS oof
```
Does work

**Comment by MinePro120** [comment]
I didn't experience that one, neither I remeber the reason I used 'echo'. But if it works...

**Comment by MinePro120** [comment]
<code> test ] </code>

**Comment by WisseHesNL** [comment]
But doesn't work

**Comment by WisseHesNL** [comment]
What would you use instead of 'echo'?

**Comment by MinePro120** [comment]
Simply cat >rds_ctl foo

**Comment by WisseHesNL** [comment]
```
pi@pi-fm:~/PiFmRds/src $ cat >rds_ctl PS foo
cat: PS: No such file or directory
cat: foo: No such file or directory
```

**Comment by WisseHesNL** [comment]
using 'PS foo' also does not work btw

**Comment by MinePro120** [comment]
Now I remeber the reason I used echo. But I didn't encounter any peoblem neither with the echo command nor with the script anyways. What RDS receiver are you using? I used my phone and noticed some latency and other issues. But that was the case with every station.

**Comment by WisseHesNL** [comment]
I tried it on my Pure Evoke H4 radio and my old Sony ICF-M50RDS. But i dont even see anything in the terminal like `PS set to:`

**Comment by MinePro120** [comment]
So the whole script doesn't work at all?

**Comment by WisseHesNL** [comment]
I think so, but i tried other things like 
```
cat >rds_ctl <<EOF
PS TEST
EOF
```
And it does change the text in the FIFO file, but PiFmRds doesn't see it for some reason.

**Comment by MinePro120** [comment]
I don't know, it worked in my case. Maybe @ChristopheJacquet can help.

**Comment by WisseHesNL** [comment]
Did you use Pi_fm_adv or just PiFmRds because Miegl said you have to run it in the pi_fm_adv directory

**Comment by MinePro120** [comment]
What is Pi_fm_adv? I ran it inside PiFmRfs/src.

**Comment by WisseHesNL** [comment]
Oh okay, Miegl made a more advanced version of PiFmRds

**Comment by miegl** [comment]
Can confirm; PiFmRds isn't able read from the pipe after a while. Haven't figured out why yet.

**Comment by MinePro120** [comment]
Oh wait, I just remembered that I've been using F5OEO's version of PiFmRds. Oops.

**Comment by WisseHesNL** [comment]
THANKS!! It works!!

**Comment by MinePro120** [comment]
What works?

**Comment by WisseHesNL** [comment]
The script does work with F5OEO's version of PiFmRds :)

**Comment by MinePro120** [comment]
And you gained some sound quality improvements too. But under no circumstances should you put any antenna on that thing. I even built a low-pass passive filter, but it was in vain.

**Comment by WisseHesNL** [comment]
(i'm a complete noob) Why should you not put an antenna on it?

**Comment by MinePro120** [comment]
Because obviously Raspberry Pi is not made for this. The sine fm radio waves are generated by the pwm function of the pi, which creates huge harmonics, namely other waves at many different frequencies. I did some tests myself. When I set the frequency to 108 Mhz, there were signals at 200 and 300 Mhz stronger than the original! I also found the range to be ~500m with a whip antenna, simply a wire connected to the pin, without any obstacles. So it is certain that it will interfere with other frequencies, possibly aircraft communications too. Not a good idea to leave it running for a long time.
